Source/Core/Core/IOS/Network/Socket.cpp

```diff
@@ -604,23 +605,35 @@ void WiiSocket::DoSock(Request request, SSL_IOCTL type)
   pending_sockops.push_back(so);
 }
 
-void WiiSockMan::AddSocket(s32 fd)
+s32 WiiSockMan::AddSocket(s32 fd)
 {
   if (fd >= 0)
   {
-    WiiSocket& sock = WiiSockets[fd];
+    s32 wii_fd = 3;
```

Comment:
You should just be able to look at IOS disasm to figure this out, which would give a better understanding anyways.